src/ipa_pc/mod.rs

```diff
@@ -511,7 +511,7 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
             .zip(states)
             .map(|((((commitments, query_set), values), proof), state)|
                 {
-                    let mut fs_rng = FiatShamirChaChaRng::<D>::from_seed(&Self::PROTOCOL_NAME);
+                    let mut fs_rng = FiatShamirChaChaRng::<D>::default();
```

Comment:
There is probably a reason for this design decision: passing the states rather than function pointers to the Fiat-Shamir random oracle.